util-core/src/test/scala/com/twitter/util/FutureTest.scala

```diff
@@ -308,6 +308,65 @@ class FutureTest extends WordSpec with MockitoSugar with GeneratorDrivenProperty
           }
         }
 
+        "execute when flushBatch is called" in {
+          val f = mock[Seq[Int] => Future[Seq[Int]]]
+          val batcher = Future.batched(4)(f)
+
+          batcher(1)
+          batcher(2)
+          batcher(3)
+          batcher.flushBatch()
+
+          verify(f).apply(Seq(1,2,3))
+        }
+
+        "only execute for remaining items when flushBatch is called after size threshold is reached" in {
+          val f = mock[Seq[Int] => Future[Seq[Int]]]
+          val batcher = Future.batched(4)(f)
+
+          batcher(1)
+          batcher(2)
+          batcher(3)
+          batcher(4)
+          batcher(5)
+          batcher.flushBatch()
+
+          verify(f, times(1)).apply(Seq(1,2,3,4))
```

Comment:
can you move the first verify (ln 334) above the flushBatch call? that would make the behavior clearer because as-is it looks like neither batch is dispatched until flushing.